src/sage/misc/cachefunc.pyx

```diff
@@ -864,11 +863,31 @@ cdef class CachedFunction():
         """
         return _cached_function_unpickle, (self.__cached_module__, self.__name__, self.cache)
 
-    #########
-    #  Introspection
+    def is_pickled_with_cache(self):
```

Comment:
Since the `do_pickle` property is public (it is not an underscored name) I do not think this method is needed. Just access `.do_pickle` directly.